### PR TITLE
fix(codex-account): a legacy auth file written a moment ago is not "from the future"

### PR DESCRIPTION
**Pull request by NeatCode-Labs**
Base: `origin/main`. Independent of every other open PR. Two commits.

## Problem

`isLegacyAuthFileFresh` rejects any timestamp later than `Date.now()`. When the in-file `last_refresh` is absent the timestamp is the file's mtime, and a file written a moment ago can carry an mtime a little ahead of `Date.now()`: on some hosts filesystem timestamps round up to their own granularity and are not read from the same clock. The freshly written legacy `auth.json` then counts as stale, the ChatGPT account is reported absent, and the suite's own "falls back to legacy auth.json when the accounts registry is absent" case fails intermittently on CI runners (seen on a Linux `test (1/2)` shard of #542, a PR that does not touch this code).

## Fix

A lead of up to five seconds is tolerated when deciding freshness. A timestamp genuinely in the future is still unknown, hence stale, which keeps the original reason for the rule (never reuse a refresh token whose age cannot be trusted).

## Tests

`detectCodexLocalAccountArtifacts.test.ts`: an `auth.json` whose mtime is one second ahead of the clock is fresh (fails on `main`, passes here); one whose mtime is a minute ahead is still stale (negative control, mirrors the existing case for a `last_refresh` a minute ahead, which keeps passing). 20/20 in the file. `pnpm typecheck`, size guard, `eslint` and `prettier --check` on the changed file.

## Review follow-ups

- *Add tests for the configured skew boundary* (review comment on the test file): valid, the two mtime cases sat at +1s and +60s, a second inside and 55 seconds outside a five-second window, so the window itself was never asserted. `test(codex-account): pin the legacy auth mtime tolerance at its boundary` adds two cases one millisecond apart on the boundary. The clock is stubbed relative to the file's own mtime rather than to wall time, so elapsed test time cannot move the classification: exactly the tolerance is fresh, because the comparison is `<=`, and one millisecond more is stale. Mutation-checked in both directions - tightening the comparison to `<` fails the first case, widening the constant fails the second.

## Tested on

Windows 11 Pro, from source, against `origin/main @ 07be9b5fa`. The intermittent failure itself was observed on the upstream Linux CI runner; locally on Windows the timestamps happened to agree, which is why the flake never reproduced here. Not run on macOS or Linux locally.

---
**Authorship & provenance:** All code in this PR was written by **Claude (Fable 5)**, directed and live-verified on Windows by me. As with anything under AGPL-3.0, it is offered **as-is, without warranty of any kind** - use, adapt, or reject freely.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved detection of recently refreshed local authentication files when their timestamps are slightly ahead due to minor system-clock differences.
  - Authentication files up to 5 seconds ahead of the current time are now recognized as fresh, while significantly older or more future-dated files remain stale.

### CODE REVIEW

File: test/features/codex-account/main/infrastructure/detectCodexLocalAccountArtifacts.test.ts [hunk 1]
```diff
@@ -302,6 +302,41 @@ describe('detectCodexLocalAccountArtifacts', () => {
...
+  it('still treats a legacy auth.json whose mtime is well in the future as stale', async () => {
+    const { codexHome, accountsDir } = await makeCodexHome();
+    const legacyAuthPath = path.join(codexHome, 'auth.json');
+    await writeFile(
+      legacyAuthPath,
+      JSON.stringify({ auth_mode: 'chatgpt', tokens: { refresh_token: 'legacy-refresh-token' } }),
+      'utf8'
+    );
+    const farAhead = new Date(Date.now() + 60_000);
+    await utimes(legacyAuthPath, farAhead, farAhead);
```
[the remark this answers is not in the document]
Comment (NeatCode-Labs) [reply test/features/codex-account/main/infrastructure/detectCodexLocalAccountArtifacts.test.ts:353]: Fixed in 95d2c267e. The two mtime cases sat at +1s and +60s, a second inside and 55 seconds outside the window, so the 5,000 ms contract itself was never asserted; the file now also has "treats a legacy auth.json whose mtime leads the clock by exactly the tolerated skew as fresh" and "treats a legacy auth.json whose mtime leads the clock by one millisecond past the tolerated skew as stale". Both use a controlled clock pinned to the file's own mtime, read back with stat after the write, and stub Date.now to that mtime minus the tolerance and minus one millisecond more, so elapsed test time cannot change the classification and the two cases differ by exactly one millisecond. isLegacyAuthFileFresh compares lastRefreshMs - now <= LEGACY_AUTH_FUTURE_SKEW_TOLERANCE_MS, so the bound is inclusive and the first case documents that. Both were mutation-checked: tightening the comparison to < fails the boundary case, widening the constant to 6_000 fails the one-millisecond-past case.
